yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -490,9 +498,11 @@ private[spark] class ApplicationMaster(
         new MutableURLClassLoader(urls, Utils.getContextOrSparkClassLoader)
       }
 
+    var userArgs = args.userArgs
     if (args.primaryPyFile != null && args.primaryPyFile.endsWith(".py")) {
-      System.setProperty("spark.submit.pyFiles",
-        PythonRunner.formatPaths(args.pyFiles).mkString(","))
+      // Second argument is the list of files to add to PYTHONPATH, which Client.scala already
+      // handles, so it's empty.
```

Comment:
It took a while for me to get this. This is actually referring to the arguments to `o.a.s.deploy.PythonRunner`'s main method. We need to document this because others who are not familiar with Spark submit at all will most certainly get lost around here.